### ISSUE TITLE
[3.17] libtiff.lib  Il mismatch between 'P1' version '20130802' and 'P2' version '20100826'

### ISSUE THREAD

**Issue by tankorsmash** [cocos2d/cocos2d-x#18657]
- cocos2d-x version:
3.17 (pulled v3 as of yesterday)
- devices test on:
win32 on Windows 10
- developing environments
   - NDK version:
   - Xcode version:
   - VS version: v12 VS2013
   - browser type and version:

Steps to Reproduce:

1.  `cocos new CCX_317_Test_2 -p com.TestReplace.CCX_317_Test_2 -l cpp`
2. `cocos compile -s .\CCX_317_Test_2 -p win32 -m release -o .\CCX_317_Test_2\bin`
3. Output:

```
3>  4 File(s) copied
3>libtiff.lib(tif_win32.obj) : warning LNK4075: ignoring '/EDITANDCONTINUE' due to '/OPT:ICF' specification
3>LINK : fatal error C1900: Il mismatch between 'P1' version '20130802' and 'P2' version '20100826'
3>LINK : fatal error LNK1257: code generation failed
4>------ Build started: Project: CCX_317_Test_2, Configuration: Release Win32 ------
4>C:\Program Files (x86)\MSBuild\12.0\bin\Microsoft.Common.CurrentVersion.targets(1492,5): warning : The referenced project '..\cocos2d\external\Box2D\proj.win32\libbox2d.vcxproj' does not exist.
4>  AppDelegate.cpp
4>  HelloWorldScene.cpp
4>  main.cpp
4>LINK : fatal error LNK1181: cannot open input file 'C:\Users\Username\Documents\cocos_projects\CCX_317_Test_2\proj.win32\Release.win32\libcocos2d.lib'
```

This is a totally fresh project I'm trying to get started, as I'm moving my old project from 3.10 to 3.17. I'll try to use 3.16 instead after this and see if that works.

There's probably something on my end, but if I can help at all, let me know.

**Comment by tankorsmash** [comment]
Closing for now, as I realized 3.17 isn't out yet.

**Comment by tankorsmash** [comment]
Once this happened on 3.16 too, I went and really made sure what I was doing. It turns out I was skipping a step where I didn't run `download-deps.py` properly and ended up copying over the lib from 3.10 instead (which was a mistake). 

Once I ran download-deps, the external folder contained the missing libbox2d.sln file which then built the libbox2d.lib I was missing, which works with the current version of the cocos2d libtiff version.

Thanks for rubber ducking!
